### ISSUE TITLE
Hacky dealing with cmake generated flags

### ISSUE THREAD

**Issue by niosus** [niosus/EasyClangComplete#123]
Forcing the last pull request there was a hacky change I did. It probably has to be fixed. The line I refer to is here: https://github.com/niosus/EasyClangComplete/commit/65120750c22c22e07bf66c528d24a73b45617802#diff-f3fcad4baaedda0d503e4b04699b7927R354

**Comment by rchl** [comment]
You mentioned that we currently take all arguments from the file. Is that opposed to only selected ones (includes and defines) previously? If so, then I would argue that taking all arguments is the proper behavior. With the proper I'm working on, if I would be using compiler_options file, I would want to use all compiler options as all are really needed to properly compile some compilation units. What would be the advantage of filtering the arguments really?

Ignore me if I misunderstood and you are talking about something else.

**Comment by niosus** [comment]
Borderline. I too think we should handle all flags. Before I didn't. Now I do, but I manually add a "-" in front of them as I split them in compilation database on `" -"` (`<space><minus>`). I think this is quite hacky and that is what I would like to fix at some point.

**Comment by rchl** [comment]
Maybe just using shlex.split would work instead?

**Comment by niosus** [comment]
That is a good idea. It's nice to have somebody around who knows which cool tools are available in Python. I am a total noob here :)

I will try in during the week probably.

**Comment by niosus** [comment]
A quick test shows that is separates smth like `-isystem <path>` into two tokens. But maybe one could post-process it.

**Comment by rchl** [comment]
But that is not a problem. It was separated in this invocation - no problem:
```
DEBUG:EasyClangComplete.plugin.completion.lib_complete: clang flags are: ['-c', '-fsyntax-only', '-x', 'c++', '-std=c++11', '-isystem', '/usr/include', '-I/Users/rchlodnicki/Dropbox/sublime/Packages/EasyClangComplete/src', '-I/workspace/easy_clang_complete/src', '-I/usr/lib/clang/3.8/include', '-I/Users/rchlodnicki/Dropbox/sublime/Packages/EasyClangComplete/tests', '-I/Users/rchlodnicki/Dropbox/sublime/Packages/EasyClangComplete', '-I/home/igor/.config/sublime-text-3/Packages/EasyClangComplete/src blah', '-I/home/igor/.config/sublime-text-3/Packages/EasyClangComplete', '-I/usr/include/opencv']
DEBUG:EasyClangComplete.plugin.completion.lib_complete: compilation started for view id: 119
DEBUG:EasyClangComplete.plugin.completion.lib_complete: compilation done in 0.05859994888305664 seconds
```

**Comment by niosus** [comment]
Yes, now, after #135 is fixed, this is not an issue.

**Comment by rchl** [comment]
Yes, but I've purposely reverted that fix to test if separating that flag breaks anything. It doesn't.

**Comment by niosus** [comment]
This is strange. The space between -isystem and actual flag doesn't work for me. Introducing the space instantly fixes everything

**Comment by rchl** [comment]
How does the clang command look like when it doesn't work? As I've pasted before, for me it is `'-isystem', '/usr/include',` and works (tried on Mac).

**Comment by niosus** [comment]
I had them merged as `-isystem /usr/include` on Ubuntu.
